Fixed typo in cube description (#374)
polycube-network/polycube: Documentation/cubes.rst

A ``Cube`` is a logical entity, composed by a ``data plane`` (which can include one or more eBPF programs) and a ``control/management plane``.

In addition, a Cube may include also a ``slow path`` in case some additional data plane-related code is required to overcome some limitations of the eBPF data plane.
This is the case of flooding packets in an 802.1 bridge, when the MAC destination address does not exist (yet) in the filtering database.

> In fact, this action (which requires a packet to be sent on all active ports except the one on which it was received) involves redirecting a packet on multiple interfaces, an operation that is not supported by the eBPF technology.

Author: ```suggestion
In fact, this action (which requires a packet to be sent on all active ports except the one on which it was received) involves redirecting a packet on multiple interfaces, an operation that is not supported by the eBPF technology.
In addition, this operation may require a loop in the dataplane, i.e., cycling on the list of interfaces and send them the packet, which is hard to do due to the eBPF limitations.
```

> In fact, this action (which requires a packet to be sent on all active ports except the one on which it was received) involves a loop, which may not be easy to achieve in the data plane given the limitations of the eBPF technology.

Reviewer: ```suggestion
In fact, this action (which requires a packet to be sent on all active ports except the one on which it was received) involves redirecting a packet on multiple interfaces, an operation that is not supported by the eBPF technology.
```

A transparent cube is a cube without any forwarding capability, such as a network monitor, NAT and a firewall.

A transparent cube:
  - does not define any port, hence it cannot be *connected* to other services, but *attached* to an existing port;
  - it can be *attached* to an existing port such as the one of a normal service (e.g., port1 on router2) or a network device (netdev) in the host (e.g., `veth0` or `eth0`);
  - it *inherits* all the parameters associated to ports (e.g., MAC/IPv4 addresses, link speed, etc.) from the actual port it is attached to;

>   - it allows to have a stack of transparent services on top of a port, very similar to a stack of functions.

Author: ```suggestion
  - it allows to have a *stack* of transparent services on top of a port, very similar to a stack of functions.
```

A transparent cube can define two processing handlers, *ingress* and *egress*, which operate on two possible traffic directions:

>   - *ingress*: it handles the traffic that goes *toward* the port it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the TCP/IP stack and it is expected to *leave* the host from that port);
>   - *egress*: it handles the traffic that comes from the cube/netdev it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the external world and that is expected to go toward the TCP/IP stack of the host).

Reviewer: ```suggestion
  - *ingress*: it handles the traffic that goes *toward* the port it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the external world and that is expected to go toward the TCP/IP stack of the host);
  - *egress*: it handles the traffic that comes from the cube/netdev it is attached to (e.g., in case of an `eth0` netdev, the traffic that comes from the TCP/IP stack and it is expected to *leave* the host from that port).
```
